src/git.zsh

```diff
@@ -479,22 +478,29 @@ _fzf_complete_git_parse_completing_option() {
     esac
 }
 
+_fzf_complete_remove_quotes_while_holding_emtpy() {
+    local tmp=(\\0${(Q)^${(z)@}})
+    echo ${(q+)tmp#\\0}
+}
+
 _fzf_complete_git_parse_argument() {
-    local arguments=(${(z)1})
+    local arguments=($(_fzf_complete_remove_quotes_while_holding_emtpy $1))
```

Comment:
I don't think `_fzf_complete_remove_quotes_while_holding_emtpy` is necessary. Why not just like:
```suggestion
    local arguments=(${(z)1})
```

Reply:
If it is `${(z)1}`, the function behaviour isn't correct when the options are quoted as follow:
```sh
$ git pull '--prune' **<TAB>

# Expected: repository list
# Actual: ref list
```
Using `${(Q)${(z)1}}` solves it, but the function hehaviour isn't correct when empty string exsits as follow:
```sh
$ git pull '' **<TAB>

# Expected: ref list
# Actual: repository list
```

Reply:
Unquote each value each time it uses an element in that array.